src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -197,46 +204,117 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
+		if (foldersRequested) {
+			resourceCompletions.push({
+				label: lastWordFolder.length === 0 ? '.' : lastWordFolder,
+				provider,
+				kind: TerminalCompletionItemKind.Folder,
+				isDirectory: true,
+				isFile: false,
+				detail: basename(cwd.fsPath),
+				replacementIndex: cursorPosition - lastWord.length,
+				replacementLength: lastWord.length
+			});
+		}
+
+		// Handle absolute paths differently to avoid adding `./` prefixes
+		const isAbsolutePath = isWindows
+			? /^[a-zA-Z]:\\/.test(lastWord)
+			: lastWord.startsWith(resourceRequestConfig.pathSeparator) && lastWord.endsWith(resourceRequestConfig.pathSeparator);
+
+		// Add all direct children files or folders
+		//
+		// For example:
+		// - `cd ./src/` -> `cd ./src/folder1`, ...
+		if (!isAbsolutePath) {
+			for (const stat of fileStat.children) {
+				let kind: TerminalCompletionItemKind | undefined;
+				if (foldersRequested && stat.isDirectory) {
+					kind = TerminalCompletionItemKind.Folder;
 				}
-				if (lastWord.length && lastWord.at(-1) !== resourceRequestConfig.pathSeparator && lastWord.at(-1) !== '.') {
-					label = `.${resourceRequestConfig.pathSeparator}${fileName}`;
+				if (filesRequested && !stat.isDirectory && (stat.isFile || stat.resource.scheme === Schemas.file)) {
+					kind = TerminalCompletionItemKind.File;
 				}
+				if (kind === undefined) {
+					continue;
+				}
+				const isDirectory = kind === TerminalCompletionItemKind.Folder;
+				const resourceName = basename(stat.resource.fsPath);
+
+				// Add a . to the start of the path if there isn't one already
+				let label = !/^\.\.?\//.test(lastWordFolder) ? `.${resourceRequestConfig.pathSeparator}${resourceName}` : `${lastWordFolder}${resourceName}`;
+
+				// Ensure directories end with a path separator
+				if (isDirectory && !label.endsWith(resourceRequestConfig.pathSeparator)) {
+					label = `${label}${resourceRequestConfig.pathSeparator}`;
+				}
+
+				// Normalize path separator to `\` on Windows. It should act the exact same as `/` but
+				// suggestions should all use `\`
+				if (isWindows) {
+					label = label.replaceAll('/', '\\');
+				}
+
+				resourceCompletions.push({
+					label,
+					provider,
+					kind,
+					isDirectory,
+					isFile: kind === TerminalCompletionItemKind.File,
+					replacementIndex: cursorPosition - lastWord.length,
+					replacementLength: lastWord.length
+				});
 			}
-			if (isDirectory && !label.endsWith(resourceRequestConfig.pathSeparator)) {
-				label = label + resourceRequestConfig.pathSeparator;
-			}
+		}
+
+		// Add parent directory to the bottom of the list because it's not as useful as other suggestions
+		//
+		// For example:
+		// - `|` -> `../`
+		// - `./src/|` -> `./src/../`
+		//
+		// On Windows, the path seprators are normalized to `\`:
+		// - `./src/|` -> `.\src\..\`
+		if (foldersRequested) {
+			const parentDir = URI.joinPath(cwd, '..' + resourceRequestConfig.pathSeparator);
 			resourceCompletions.push({
```

Comment:
Should we block parent dir for `isAbsolutePath` too? `/../` is an invalid suggestion and `/usr/../` is a strange suggestion

Reply:
Just seeing this, yep, makes sense, will do